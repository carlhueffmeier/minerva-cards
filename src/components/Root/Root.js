import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Inbox, DeckDetails, AddNote } from 'components';

function Root(props) {
  return (
    <Router>
      <div className="app">
        <h1>Minerva Cards</h1>
        <Route exact path="/" component={Inbox} />
        <Route exact path="/decks/:deckId" component={DeckDetails} />
        <Route exact path="/decks/:deckId/add-note" component={AddNote} />
        <Route exact path="/notes/add-note" component={AddNote} />
      </div>
    </Router>
  );
}

export default Root;
