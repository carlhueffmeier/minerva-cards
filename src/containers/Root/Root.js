import React from 'react';

function Root(props) {
  return (
    <div className="App">
      <h1>Minerva Cards</h1>
    </div>
  );
}

export default Root;
