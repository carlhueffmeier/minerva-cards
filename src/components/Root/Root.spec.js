import React from 'react';
import { shallow } from 'enzyme';
import Root from './Root';

describe('Root', () => {
  const root = shallow(<Root />);

  it('renders properly', () => {
    expect(root).toMatchSnapshot();
  });
});
