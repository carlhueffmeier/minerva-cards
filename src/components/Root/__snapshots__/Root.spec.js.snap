// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Root />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
>
        <h1>
                Minerva Cards
        </h1>
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/decks/:deckId"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/decks/:deckId/add-note"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/notes/add-note"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Minerva Cards
</h1>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/decks/:deckId"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/decks/:deckId/add-note"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/notes/add-note"
/>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Minerva Cards",
          },
          "ref": null,
          "rendered": "Minerva Cards",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/decks/:deckId",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/decks/:deckId/add-note",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/notes/add-note",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
>
          <h1>
                    Minerva Cards
          </h1>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/decks/:deckId"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/decks/:deckId/add-note"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/notes/add-note"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Minerva Cards
</h1>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/decks/:deckId"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/decks/:deckId/add-note"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/notes/add-note"
/>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Minerva Cards",
            },
            "ref": null,
            "rendered": "Minerva Cards",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/decks/:deckId",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/decks/:deckId/add-note",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/notes/add-note",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
